robottelo/host_helpers/contenthost_mixins.py

```diff
@@ -109,11 +110,9 @@ def _dogfood_helper(self, product, release, repo=None):
     def download_repofile(self, product=None, release=None, snap='', proxy=None):
         """Downloads the tools/client, capsule, or satellite repos on the machine"""
         product, release, v_major, _ = self._dogfood_helper(product, release)
-        if not proxy and settings.server.is_ipv6:
```

Comment:
It looks like we're just removing the ability to pass the proxy to this function. If that's intentional, we should also remove the proxy argument from the function def too,

Reply:
Thanks for spotting it, fixed in c132a7b80923384f43741d700e1832b07fcf5e51

Reply:
@ogajduse I think here and at many places we would be configuring proxy required for IPv6 only for dualstack as well, which would be unnecessary for dualstack, so I'd suggest to have a new property like `is_ipv6` and use that as a condtion to configure proxy, wdyt?

Reply:
"many places" - yes. That is solvable by `sat_host.network_type.has_ipv4`. Please take a look at how it is being used in this PR.

However, we do not need to set the proxy up here since robottelo runs on infrastructure where both IP stacks are configured.

One of the goals of this PR is to get rid of `is_ipv6` logic - that should be replaced by using the new enum.
Will this change introduce any inconvenience for robottelo users?

Reply:
cc @JacobCallahan , any thoughts?

Reply:
I completely agree with your statement. I did not realize the scenario when the outbound connections to the internet from our IPv6 network need to be proxied. 

As this PR became time-sensitive, I will proceed with merging it. We're taking the risk of breaking some test cases. 

The means to distinguish the network type is still there via the `network_type` host property, therefore, it could be used in the same way as `server.is_ipv6` was.